server/src/main/java/io/crate/replication/logical/MetadataTracker.java

```diff
@@ -225,12 +227,38 @@ private CompletableFuture<?> processSubscription(String subscriptionName) {
                     subscription.connectionInfo().user()
                 );
                 return client.execute(PublicationsStateAction.INSTANCE, request)
-                    .thenCompose(publicationsStateResp -> subscribeToNewRelations(
-                        subscriptionName,
-                        subscription,
-                        subscriberState,
-                        publicationsStateResp,
-                        replicationService));
+                    .thenCompose(publicationsStateResp -> {
+                        // We cannot use replicationService.verifyTablesDoNotExist(subscriptionName, stateResponse)
```

Comment:
> within the updateClusterState routine

not sure. for this check we need `PublicationsStateAction.Response` which we don't have before the tracking begins for example, here https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/replication/logical/LogicalReplicationService.java#L203 this check would be irrelevant if it's inlined into `updateSubscriptionState`

Reply:
In any case, leaving the cluster state in a broken state is not an option. I think it might be possible to replace the cluster state retrieval by getting all requires information via `PublicationsStateAction.Response`. I'll have a closer look at that option.

Reply:
I updated it to do the check before `updateClusterState` - with the new fetch structure it's possible. But same result - passes if I tweak `publisherConnectionUrl` to always use sniff and fails if I do other way around (pg_tunnel) 😞

Reply:
I updated it to do the check before `updateClusterState` - with the new fetch structure it's possible. But same result - passes if I tweak `publisherConnectionUrl` to always use sniff and fails if I do other way around (pg_tunnel) 😞

Reply:
Hi @mfussenegger, thanks for fixing deadlock - test passes now (2 & 3 commits). Could you please take another look?